catalog/imaging/imaging-postprocess-plugin-nitf/src/test/groovy/org/codice/alliance/plugin/nitf/NitfPostProcessPluginSpec.groovy

```diff
@@ -0,0 +1,415 @@
+import org.codice.imaging.nitf.render.NitfRenderer
+import spock.lang.Ignore
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import java.awt.image.BufferedImage
+import java.util.function.Supplier
+
+class NitfPostProcessPluginSpec extends Specification {
+
+    private static final String GEO_NITF = "/i_3001a.ntf"
+
+    private static final long TEST_SIZE = 5L * 1024L * 1024L
+
+    private static final double DEFAULT_MAX_SIDE_LENGTH = 1024.0
+
+    private NitfPostProcessPlugin nitfPostProcessPlugin = null
+
+    private ProcessRequest<ProcessCreateItem> createProcessRequest
+
+    private ProcessRequest<ProcessUpdateItem> updateProcessRequest
+
+    private Metacard metacard
+
+    private Metacard metacard1
+
+    private ProcessResource processResource
+
+    private ProcessResource processResource1
+
+    private ProcessCreateItem processCreateItem
+
+    private ProcessCreateItem processCreateItem1
+
+    private ProcessUpdateItem processUpdateItem
+
+    private ProcessUpdateItem processUpdateItem1
+
+    private Supplier<NitfRenderer> rendererSupplier
+
+    def setup() {
+        nitfPostProcessPlugin = new NitfPostProcessPlugin({
+            return Mock(NitfRenderer) {
+                final BufferedImage bufferedImage = Mock(BufferedImage)
+
+                render() >> bufferedImage
+                renderToClosestDataModel() >> bufferedImage
+            }
+        })
+
+        metacard = Mock(Metacard) {
+            getId() >> "0000000"
+        }
+
+        metacard1 = Mock(Metacard) {
+            getId() >> "1111111"
+        }
+
+        def (resourceUri, resourceInputStream) = getTestResource(GEO_NITF)
+        processResource = Mock(ProcessResource) {
+            getUri() >> resourceUri
+            getInputStream() >> resourceInputStream
+            getMimeType() >> NitfPostProcessPlugin.NITF_MIME_TYPE.toString()
+            getSize() >> TEST_SIZE
+        }
+
+        processCreateItem = Mock(ProcessCreateItem) {
+            getMetacard() >> metacard
+            getProcessResource() >> processResource
+        }
+
+        processUpdateItem = Mock(ProcessUpdateItem) {
+            getMetacard() >> metacard
+            getProcessResource() >> processResource
+            getOldMetacard() >> Mock(Metacard)
+        }
+
+        createProcessRequest = Mock(ProcessRequestImpl) {
+            getProcessItems() >> [processCreateItem]
+        }
+
+        updateProcessRequest = Mock(ProcessRequestImpl) {
+            getProcessItems() >> [processUpdateItem]
+        }
+    }
+
+    @Unroll
+    def "Building derived image filename from \"#ftitle\"" (String ftitle, String expectedFname) {
+        setup:
+            NitfPostProcessPlugin plugin = new NitfPostProcessPlugin(rendererSupplier);
+            def qualifier = "original";
+
+        when: "building a derived image filename"
+            def derivedFname = plugin.buildDerivedImageTitle(ftitle, qualifier, "jpg");
```

Comment:
✏️ Do you need to instantiate a `NitfPostProcessPlugin` here?

```groovy
    @Unroll
    def "Building derived image filename from \"#ftitle\"" (String ftitle, String expectedFname) {
        expect: "the derived filename should not include invalid characters"
        NitfPostProcessPlugin.buildDerivedImageTitle(ftitle, "original", "jpg") == expectedFname

        where:
            ftitle                                                     ||  expectedFname
            null                                                       ||  "original.jpg"
            ""                                                         ||  "original.jpg"
            "_"                                                        ||  "original.jpg"
            "@#\$%^&*()+-={}|[]<>?:"                                   ||  "original.jpg"
            "Too Legit To Quit"                                        ||  "original-toolegittoquit.jpg"
            "A bunch of _invalid_ characters! @#\$%^&*()+-={}|[]<>?:;" ||  "original-abunchof_invalid_characters.jpg"
    }
```

Reply:
Fixed